spec/helpers/result_filters/subject_helper_spec.rb

```diff
@@ -12,4 +12,16 @@
       expect(helper.subject_is_selected?(id: "4")).to eq(false)
     end
   end
+
+  describe "#no_subject_selected_error?" do
+    it "returns true if no subject is selected" do
+      allow(controller).to receive(:flash).and_return(error: I18n.t("subject_filter.errors.no_option"))
+
+      expect(helper.no_subject_selected_error?).to eq(true)
+    end
+
+    it "returns nil if there is no subject selected" do
+      expect(helper.no_subject_selected_error?).to eq(nil)
```

Comment:
> Should no_subject_selected_error? return false instead of nil?

Very good question...

I say it should just really be `true|false` as there isn't much to it and there is nothing can be operated in the whatever object that it can returned, in either case, I'm not fussed.

Nice that you made a test of it to illustrate it. :+1: 


@misaka your call or anyone else for final verdict/team vote.
![image](https://user-images.githubusercontent.com/470137/74553251-dd799e80-4f4e-11ea-8c21-b21f1330f887.png)

Reply:
The method is a question so I would expect true or false